ReplicatorHelper.swift

```diff
@@ -8,6 +8,103 @@ import JavaScriptCore
 import CouchbaseLiteSwift
 
 public struct ReplicatorHelper {
+    private static let jsContext: JSContext = {
+        let context = JSContext()!
+        
+        let logFunction: @convention(block) (String) -> Void = { message in
+            print("JSFilter log: \(message)")
+        }
+        
+        context.setObject(logFunction, forKeyedSubscript: "log" as NSString)
+        context.evaluateScript("var console = { log: log };")
+        
+        // Handle JavaScript exceptions
+        context.exceptionHandler = { context, exception in
+            if let exc = exception {
+                print("JSFilter Exception: \(exc.toString() ?? "unknown error")")
+            }
+        }
+        
+        return context
+    }()
+    
+  private static func evaluateFilter(_ filterFunction: String, document: Document, flags: DocumentFlags) -> Bool {
+    print("FILTER DEBUG: Evaluating filter for document ID: \(document.id)")
+    
+    // Convert document to dictionary for JS
+    let docDict = document.toDictionary()
+    let jsonData = try? JSONSerialization.data(withJSONObject: docDict, options: [])
+    guard let jsonStr = jsonData.flatMap({ String(data: $0, encoding: .utf8) }) else {
+        print("Failed to convert document to JSON")
+        return false
+    }
+    
+    // Create flags object for JS
+    let flagsDict: [String: Bool] = [
+        "deleted": flags.contains(.deleted),
+        "accessRemoved": flags.contains(.accessRemoved),
+        "isDeleted": flags.contains(.deleted) // Add this for compatibility
+    ]
+    let flagsData = try? JSONSerialization.data(withJSONObject: flagsDict, options: [])
+    guard let flagsStr = flagsData.flatMap({ String(data: $0, encoding: .utf8) }) else {
+        print("Failed to convert flags to JSON")
+        return false
+    }
+    
+    // Create a wrapper function to call the filter with better logging
+    let wrapperScript = """
+    function evaluateFilter() {
+        console.log("FILTER DEBUG: Starting filter evaluation for document");
+        
+        try {
+            const filterFunc = \(filterFunction);
+            console.log("FILTER DEBUG: Filter function defined successfully");
+            
+            const doc = JSON.parse('\(jsonStr)');
+            console.log("FILTER DEBUG: Document parsed, ID: " + (doc._id || doc.id));
```

Comment:
The jsonStr will not contain `_id` property. The `id` property is not a reserved property for document id so it could be a user's defined property from the body of the document.